feeds/wlan-ap/opensync/src/src/sm/src/ubus_collector.c

```diff
@@ -541,6 +561,94 @@ static void ubus_collector_bss_cb(struct ubus_request *req, int type,
 	}
 }
 
+static void ubus_collector_chan_switch_events_cb(struct ubus_request *req, int type,
+				  struct blob_attr *msg)
+{
+
+	int error = 0;
+	int rem = 0;
+	struct blob_attr *tb_chan_event_lst[__CHANNEL_SWITCH_EVENT_MAX] = {};
+	struct blob_attr *tb_chan_event_table[__CHANNEL_SWITCH_MAX] = {};
+	struct blob_attr *tb_chan_event_tbl = NULL;
+	dpp_event_channel_switch_t *channel_event_record = NULL;
+
+	if (!msg)
+		return;
+
+	LOG(INFO, "ubus_collector: received ubus collector chan event message");
+
+	error = blobmsg_parse(ubus_collector_chan_switch_event_policy,
+			      __CHANNEL_SWITCH_EVENT_MAX, tb_chan_event_lst,
+			      blobmsg_data(msg), blobmsg_data_len(msg));
+	if (error || !tb_chan_event_lst[CHAN_SWITCH_EVENT]) {
+		LOG(INFO,
+		    "ubus_collector: failed while parsing chan event policy");
+		return;
+	}
+
+	/* itereate channel event list */
+	blobmsg_for_each_attr(tb_chan_event_tbl, tb_chan_event_lst[CHAN_SWITCH_EVENT], rem)
+	{
+		error = blobmsg_parse(channel_switch_event_policy,
+				      __CHANNEL_SWITCH_MAX, tb_chan_event_table,
+				      blobmsg_data(tb_chan_event_tbl),
+				      blobmsg_data_len(tb_chan_event_tbl));
+		if (error) {
+			LOG(INFO,
+			    "ubus_collector_ failed while parsing chan event policy");
+			continue;
+		}
+
+		channel_event_record = malloc(sizeof(dpp_event_channel_switch_t));
+
+		channel_event_record->channel_event.band = frequency_to_band(blobmsg_get_u32(tb_chan_event_table[CHANNEL_SWITCH_RADIO_NAME]));
+
+		channel_event_record->channel_event.reason = blobmsg_get_u8(tb_chan_event_table[CHANNEL_SWITCH_REASON]);
+		channel_event_record->channel_event.freq = blobmsg_get_u8(tb_chan_event_table[CHANNEL_SWITCH_FREQ]);
```

Comment:
Missing frequency to channel conversion. Further, u8 is not sufficient to hold frequency.